samples/extensions/dynamic_multisample_rasterization/dynamic_multisample_rasterization.h

```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) 2024, Mobica Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+class DynamicMultisampleRasterization : public ApiVulkanSample
+{
+  public:
+	DynamicMultisampleRasterization();
+	virtual ~DynamicMultisampleRasterization();
+
+  private:
+	std::unique_ptr<vkb::core::Buffer> vertex_buffer = nullptr;
+	std::unique_ptr<vkb::core::Buffer> index_buffer  = nullptr;
+
+	std::unique_ptr<vkb::sg::Scene>    scene;
+	std::vector<VkDescriptorImageInfo> image_infos;
+	std::map<std::string, int32_t>     name_to_texture_id;
+
+	struct SceneNode
+	{
+		vkb::sg::Node    *node;
+		vkb::sg::SubMesh *sub_mesh;
+	};
+	std::vector<SceneNode> scene_nodes_opaque;
+	std::vector<SceneNode> scene_nodes_opaque_flipped;
+	std::vector<SceneNode> scene_nodes_transparent;
+	std::vector<SceneNode> scene_nodes_transparent_flipped;
+
+	struct UniformData
+	{
+		glm::mat4 projection;
+		glm::mat4 view;
+	} uniform_data;
+	std::unique_ptr<vkb::core::Buffer> uniform_buffer;
+
+	VkPipeline            pipeline;
+	VkPipeline            pipeline_inversed_rasterizer;
+	VkPipelineLayout      pipeline_layout;
+	VkDescriptorSet       descriptor_set;
+	VkDescriptorSetLayout descriptor_set_layout;
+
+	// GUI
+	VkPipeline            pipeline_gui;
+	VkPipelineLayout      pipeline_layout_gui;
+	VkDescriptorSet       descriptor_set_gui;
+	VkDescriptorSetLayout descriptor_set_layout_gui;
+	VkDescriptorPool      descriptor_pool_gui;
+
+	ImageData    color_attachment;
+	VkClearValue clear_values[3];
```

Comment:
`clear_values[1]` is never used.
Besides that, is it worth to hold those values here?

Reply:
Decrease vector and remove declaration from header.